### PR TITLE
BQ Data Transfer: Add quickstart sample

### PR DESCRIPTION
**Pull request by tswast**
A sample showing how to use the BigQuery Data Transfer API to list
possible data sources.

### CODE REVIEW

**Comment by tswast** [comment]
Travis is failing on `go get`, but seems unrelated to this change.

```
github.com/go-sql-driver/mysql (download)
# cd /home/travis/gopath/src/github.com/go-sql-driver/mysql; git pull --ff-only
You are not currently on a branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.
    git pull <remote> <branch>
package github.com/go-sql-driver/mysql: exit status 1
```

File: bigquery/bigquery_datatransfer_quickstart/main.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	for {
+		resp, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			// TODO: Handle error.
+			log.Fatalf("Failed to list sources: %v", err)
+		}
+		// TODO: Use resp.
```
Comment (ahmetb) [inline bigquery/bigquery_datatransfer_quickstart/main.go]: looks like you're using it

File: bigquery/bigquery_datatransfer_quickstart/main.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	}
+	it := client.ListDataSources(ctx, req)
+	fmt.Println("Supported Data Sources:")
+	for {
+		resp, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			// TODO: Handle error.
```
Comment (ahmetb) [inline bigquery/bigquery_datatransfer_quickstart/main.go]: looks like you're handling it

File: bigquery/bigquery_datatransfer_quickstart/main.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		Parent: fmt.Sprintf("projects/%s", projectID),
+	}
+	it := client.ListDataSources(ctx, req)
+	fmt.Println("Supported Data Sources:")
+	for {
+		resp, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
```
Comment (ahmetb) [inline bigquery/bigquery_datatransfer_quickstart/main.go]: else if err != nil

File: bigquery/bigquery_datatransfer_quickstart/main.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		log.Fatalf("Failed to create client: %v", err)
+	}
+
+	req := &datatransferpb.ListDataSourcesRequest{
+		Parent: fmt.Sprintf("projects/%s", projectID),
+	}
+	it := client.ListDataSources(ctx, req)
+	fmt.Println("Supported Data Sources:")
+	for {
+		resp, err := it.Next()
```
Comment (jadekler) [inline bigquery/bigquery_datatransfer_quickstart/main.go]: I might rename "resp" to something that reflects the return type of `*datatransferpb.DataSource`. Maybe `ds`?

File: bigquery/bigquery_datatransfer_quickstart/main.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	if err != nil {
+		log.Fatalf("Failed to create client: %v", err)
+	}
+
+	req := &datatransferpb.ListDataSourcesRequest{
+		Parent: fmt.Sprintf("projects/%s", projectID),
+	}
+	it := client.ListDataSources(ctx, req)
+	fmt.Println("Supported Data Sources:")
+	for {
```
Comment (jadekler) [inline bigquery/bigquery_datatransfer_quickstart/main.go]: Not sure if it makes it more or less readable, but you could squish these 4 lines and pop them into the for loop for a more iterator-y feeling: `for ds, err := it.Next(); err != iterator.Done; ds, err = it.Next() {`.

I'm unsure which style these docs prefer (noob here heh).

Comment (ahmetb) [reply bigquery/bigquery_datatransfer_quickstart/main.go]: less readable for sure. I was also going to comment with this but searched the repo. it looks like for { if err != ... } is the preferred idiom currently.

Comment (jadekler) [reply bigquery/bigquery_datatransfer_quickstart/main.go]: 👍

File: bigquery/bigquery_datatransfer_quickstart/main.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+		ds, err := it.Next()
+		if err == iterator.Done {
+			break
+		} else if err != nil {
```
Comment (broady) [inline bigquery/bigquery_datatransfer_quickstart/main.go]: nit: no need for the else statement here.

Otherwise LGTM

Comment (tswast) [reply bigquery/bigquery_datatransfer_quickstart/main.go]: Cool, that's what I had originally. :-)

https://github.com/GoogleCloudPlatform/golang-samples/pull/362#discussion_r159347062

I'm happy to change it back.

**Pull request merged**